src/nomad_catalysis/schema_packages/catalysis.py

```diff
@@ -394,3 +412,1107 @@ def normalize(self, archive, logger):
+                        / self.reactor_filling.catalyst_mass
+                    )
+
+    def return_conversion_results(
+        self, archive: 'EntryArchive', logger: 'BoundLogger'
+    ) -> None:
+        """
+        This function returns the conversion results of the reactants for the results
+        section of the archive.
+        It checks if the the name of the reactant is not in the list of the inert gases
+        and if the name is in the list of the reaction_conditions.reagents, it will try
+        to replace the name of the reactant with the IUPAC name of the reagent.
+
+        return: a list of the reactants with the conversion results.
+        """
+        if self.results[0].reactants_conversions is not None:
+            conversions_results = []
+            for i in self.results[0].reactants_conversions:
+                if i.name in ['He', 'helium', 'Ar', 'argon', 'inert']:
+                    continue
+                else:
+                    for j in self.reaction_conditions.reagents:
+                        if i.name == j.name:
+                            if j.pure_component.iupac_name is not None:
+                                i.name = j.pure_component.iupac_name
+                            if i.gas_concentration_in is None:
+                                i.gas_concentration_in = j.gas_concentration_in
+                            react = Reactant(
+                                name=i.name,
+                                conversion=i.conversion,
+                                gas_concentration_in=i.gas_concentration_in,
+                                gas_concentration_out=i.gas_concentration_out,
+                            )
+                            conversions_results.append(react)
+                            if (
+                                np.allclose(
+                                    i.gas_concentration_in, j.gas_concentration_in
+                                )
+                                is False
+                            ):
+                                logger.warn(f"""Gas concentration of '{i.name}' is not
+                                            the same in reaction_conditions and
+                                            results.reactants_conversions.""")
+        return conversions_results
+
+    def check_sample(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        if self.samples:
+            if self.samples[0].lab_id is not None and self.samples[0].reference is None:
+                sample = CompositeSystemReference(
+                    lab_id=self.samples[0].lab_id, name=self.samples[0].name
+                )
+                sample.normalize(archive, logger)
+                self.samples = []
+                self.samples.append(sample)
+            if self.samples[0].reference is not None:
+                self.populate_catalyst_sample_info(archive, logger)
```

Comment:
You get the idea by now ;)
```suggestion
        if not self.samples:
            return
        if self.samples[0].lab_id is not None and self.samples[0].reference is None:
            sample = CompositeSystemReference(
                lab_id=self.samples[0].lab_id, name=self.samples[0].name
            )
            sample.normalize(archive, logger)
            self.samples = []
            self.samples.append(sample)
        if self.samples[0].reference is not None:
            self.populate_catalyst_sample_info(archive, logger)
```